WolvenKit/Views/Editors/AutomaticNodifyEditor.cs

```diff
@@ -0,0 +1,230 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using Nodify;
+using Microsoft.Msagl.Core.Geometry.Curves;
+using Microsoft.Msagl.Core.Layout;
+using Microsoft.Msagl.Core.Routing;
+using Microsoft.Msagl.Layout.Layered;
+using WolvenKit.Functionality.Interfaces;
+using System.Windows.Input;
+using Microsoft.Msagl.Layout.MDS;
+
+namespace WolvenKit.Views.Editors
+{
+    public class AutomaticNodifyEditor : NodifyEditor
+    {
+        public AutomaticNodifyEditor() : base()
+        {
+
+        }
+
+
+        protected override void OnMouseMove(MouseEventArgs e)
+        {
+            CurrentMousePosition = e.GetPosition(this);
+
+            if (CurrentMousePosition != PreviousMousePosition)
+            {
+                MouseLocation = TransformPosition(CurrentMousePosition);
+
+                if (IsMouseCaptured)
+                {
+                    // Panning
+                    if (e.MiddleButton == MouseButtonState.Pressed)
```

Comment:
Just for my understanding: are you overriding `OnMouseMove` only to change the mouse button used for panning? (from right-click to mouse-middle)
This seems like a lot of work to do just that. I will create an issue to allow configuring keybindings for common editor actions.

Reply:
That would be great! It's been a while since I wrote this, but I recall thinking it would be nice to be able to handle some custom mouse actions here as well (right-click drag, ctrl + left-click, etc).